src/NServiceBus.Core/Audit/AuditContext.cs

```diff
@@ -12,27 +14,23 @@ public AuditContext(OutgoingMessage message, string auditAddress, IBehaviorConte
             Guard.AgainstNullAndEmpty(nameof(auditAddress), auditAddress);
             Message = message;
             AuditAddress = auditAddress;
+            AuditMetadata = new Dictionary<string, string>();
```

Comment:
In theory, we could lazy initialize the dictionary on access of the setter method in case we want to save the dictionary allocation. But I'm uncertain if we ever have a case when we don't use the AuditMetadata

Reply:
Discussed and agreed to not do this since we always add to it